### PR TITLE
[WIP] Adding accessibility guidelines to button documentation

### PR DESCRIPTION
**Pull request by colinpmacarthur**
@carodew, please review. Thanks!

### CODE REVIEW

File: _elements/buttons.md [hunk 1]
```diff
@@ -70,7 +70,11 @@ lead: Use buttons to signal actions.
   </button>
   <div id="collapsible-0" aria-hidden="false" class="usa-accordion-content">
     <h4 class="usa-heading">Accessibility</h4>
-    <p>No specific accessibility guidelines.</p>
+    <ul>
+      <li>Buttons should display a "halo" when users tab to them.</li>
```
Comment (carodew) [inline _elements/buttons.md]: How about "Buttons should display a visible focus state when users tab to them." ?

Comment (colinpmacarthur) [inline _elements/buttons.md]: Fixed in c8ced10

Comment (colinpmacarthur) [inline _elements/buttons.md]: Thanks!

File: _elements/buttons.md [hunk 2]
```diff
@@ -70,7 +70,11 @@ lead: Use buttons to signal actions.
   </button>
   <div id="collapsible-0" aria-hidden="false" class="usa-accordion-content">
     <h4 class="usa-heading">Accessibility</h4>
-    <p>No specific accessibility guidelines.</p>
+    <ul>
+      <li>Buttons should display a visible focus state when users tab to them.</li>
+      <li>Whenever possible, use the <code>&lt;button&gt;</code> to display buttons rather than styling links to look like buttons. Screen readers treat buttons differently than they do links.</li>
```
Comment (maya) [inline _elements/buttons.md]: Is this accurate? I think it's perfectly acceptable to style links like buttons. Buttons don't have `href` properties, they are more for actionable forms.

**Comment by carodew** [comment]
lgtm!

File: _elements/buttons.md (hunk 2 above)
Comment (carodew) [inline _elements/buttons.md]: Apparently there is a fair amount of controversy surrounding this. @colinpmacarthur and I spoke and we thought a more cautionary, less prescriptive approach would be more fitting our 'design for flexibility' goal. But it probably deserves a mention given the parallel priority of accessibility.

Comment (maya) [inline _elements/buttons.md]: @carodew I prefer your update which explains the difference between using a button vs link in the context of screen readers. This lets the user decide which is more appropriate.

**Pull request merged**